Client style improvements

FiwareULPGC opened the pull request:
This Pull Request has several style improvements:
- Usage of DOM systax.
- Nomenclature (correct use of english, unifiying nomenclature, folder structure)
- Implementation of the module pattern.
- Deleted unused code.
- Style improvements in order to pass JSLINT validations.
- Minor fixes.

With this PR the issues #31 #20 #19 #18 #16 should be solved.

With this PR #21 is partially solved. It will be fully  resolved in an upcoming PR that adds linter tests to travis-ci.

Due to the size of this PR and following the developer guidelines ( https://forge.fiware.org/plugins/mediawiki/wiki/fiware/index.php/Developer_Guidelines ), we have rebased the PR semantically. We can reduce the number of commits if it is needed.

We plan to refactorize the modules in a smaller ones in a upcoming PR.

jmcanterafonseca commented on client/css/app.css:
```diff
@@ -1,319 +0,0 @@
-	font-weight: bold;
-}
-
-*/
```
please avoid commented code

jmcanterafonseca commented on client/css/app.css:
```diff
@@ -1,319 +0,0 @@
-  
-}
-
-.table-fixed tbody td, .table-fixed thead > tr> th {
```
nit: space between tr and '>'

jmcanterafonseca commented on client/css/app.css:
```diff
@@ -1,319 +0,0 @@
-
-#reviews_table th, #reviews_table td{
-	
-	/*max-height: 2em;*/	
```
commented code

jmcanterafonseca commented on client/index.html:
```diff
@@ -1,79 +1,71 @@
   <meta name="viewport" content="width=device-width, initial-scale=1">
   
-
-   <!-- FI-WARE styles -->
```
nit FI-WARE should be FIWARE

jmcanterafonseca commented on client/index.html:
```diff
@@ -1,79 +1,71 @@
-    <script src="js/index.js"></script>
-    <script src="js/connections_api.js"></script>
+  <!-- app JS -->
+  <script src="js/JSUtils.js"></script>
```
probably better if you name this file as 'utils.js'

jmcanterafonseca commented on client/index.html:
```diff
@@ -1,79 +1,71 @@
+  <script src="js/connectionsAPI.js"></script>
 
-<title> FIWARE TEST</title>
+<title>FIWARE Tourguide</title>
```
nit: FIWARE Tour Guide Application

jmcanterafonseca commented on client/js/JSUtils.js:
```diff
@@ -0,0 +1,121 @@
+
+*/
+
+var Tourguide = Tourguide || {};
```
I can see the good intentions on having a root module called `TourGuide`, but it is making the code difficult to read and error prone as you depend on the previous creation of an object named `TourGuide` . Why don't you just create a module named `Request` 

``` js
var Request = (function() {
   return { 
      'get': getFunction,
      'post': postFunction 
   }
})();
```

jmcanterafonseca commented on client/js/JSUtils.js:
```diff
@@ -0,0 +1,121 @@
+    xhr.send(JSON.stringify(data));
+  }
+
+  Tourguide.AJAXRequest = {
```
not very elegant nor needed if you avoid a root namespace

jmcanterafonseca commented on client/js/connectionsAPI.js:
```diff
@@ -0,0 +1,255 @@
+*/
+
+var Tourguide = Tourguide || {};
+
```
same comment as above

jmcanterafonseca commented on client/js/connectionsAPI.js:
```diff
@@ -0,0 +1,255 @@
+
+var initConnections = function() {
+  //check if user is logged in
+  Tourguide.AJAXRequest.get('http://tourguide/client/user',
```
do you really need this? an absolute URL? strange

jmcanterafonseca commented on client/js/connectionsAPI.js:
```diff
@@ -0,0 +1,255 @@
+    return;
+  }
+
+  function createAndShowMenu(userInfo) {
```
I would divide this function in two, one for building the UI and another one for showing the UI, without mixing concerns

jmcanterafonseca commented on client/js/connectionsAPI.js:
```diff
@@ -0,0 +1,255 @@
+      else {
+        showMessage('Log in required', 'alert-warning');
+      }
+    }, 500);
```
use a constant for the timeout

jmcanterafonseca commented on client/js/connectionsAPI.js:
```diff
@@ -0,0 +1,255 @@
+  };
+})(Tourguide);
+/* alerType could be alert-warning(default) or alert-danger*/
+function showMessage(message, alertType) {
```
function out of any scope

jmcanterafonseca commented on client/js/restaurantsAPI.js:
```diff
@@ -0,0 +1,1279 @@
+(function(app) {
+  //proxy should be used if the API is not in the same location as the web app.
+  var proxyURL = '';
+
```
you don't actually need that if you use CORS

jmcanterafonseca commented on client/js/restaurantsAPI.js:
```diff
@@ -0,0 +1,1279 @@
+  var proxyURL = '';
+
+  var baseURL = proxyURL + 'http://tourguide/api/orion/';
+
```
this is really strange to be harcoded

jmcanterafonseca commented on client/js/restaurantsAPI.js:
```diff
@@ -0,0 +1,1279 @@
+      var mark = {'name': restaurant.name};
+
+      //get desired attributes
+      try {
```
please avoid try catch

use 

``` js
if (restaurant.address && restaurant.address.streetAddress) 
```

jmcanterafonseca commented on client/js/restaurantsAPI.js:
```diff
@@ -0,0 +1,1279 @@
+      }
+      catch (err) {
+        console.log('Cannot get reviewCount for ' + restaurant.name);
+      }
```
avoid try catch

jmcanterafonseca commented on client/js/restaurantsAPI.js:
```diff
@@ -0,0 +1,1279 @@
+          continue;
+        }
+      }
+      catch (err) {
```
avoid try catch

jmcanterafonseca commented on client/js/restaurantsAPI.js:
```diff
@@ -0,0 +1,1279 @@
+    /* clustering approach */
+    var markerClusters = L.markerClusterGroup({showCoverageOnHover: true});
+    var markers = [];
+    for (i = 0, len = restaurantMarks.length; i < len; i++) {
```
better 

``` js
restaurantMarks.forEach(function(aMark) { })
```

jmcanterafonseca commented on client/js/restaurantsAPI.js:
```diff
@@ -0,0 +1,1279 @@
+      //add mark to map
+      restaurantMarks[i].mark = L.marker(restaurantMarks[i].coords);
+
+      var popHTML = document.createElement('DIV');
```
please create a specific function for building the ui

jmcanterafonseca commented on client/js/restaurantsAPI.js:
```diff
@@ -0,0 +1,1279 @@
+
+
+    var option;
+    for (var i = 0; i <= 5; i++) {
```
use a constant

jmcanterafonseca commented on client/js/restaurantsAPI.js:
```diff
@@ -0,0 +1,1279 @@
+
+
+  function showEditReview(reviewResponse) {
+    reviewResponse = JSON.parse(reviewResponse);
```
separate between showing and building the UI

jmcanterafonseca commented on client/stylesheets/app.css:
```diff
@@ -0,0 +1,265 @@
+  z-index:50;
+  margin-top: 50px;
+  margin-right: 10px;
+  margin-left: 50px;
```
please provide a new line at the end of the file

jmcanterafonseca commented on client/stylesheets/app.css:
```diff
@@ -0,0 +1,265 @@
+  position: fixed;
+  top:0;
+  margin-top: 45px;
+   width: 100%;
```
nit: indent

jmcanterafonseca commented on client/js/restaurantsAPI.js:
```diff
@@ -0,0 +1,1279 @@
+'use strict';
```
this source code file is too big

Why don't you divide it by functionality, for instance: 

restaurant list
restaurant data edition
reservations 

... 

or other criteria?

jmcanterafonseca commented:
good work but we certainly need more iterations

thanks!

jmcanterafonseca commented:
there are still some comments not addressed wrt to CSS and comments,

please check

FiwareULPGC committed:
Client style improvements:
 - Nomenclature changes (camelCase, use of english, renamed files, etc).
 - Translate to DOM syntax.
 - Process reviews onsubmit instead of onclick.
 - DOM reservationForm. Fixed disabled/enabled form elements.
 - Deleted commented code. Delete unnecesary code.
 - Removed several console.log lines.
 - Used innerHTML= '' instead of removeChild.
 - Module pattern implemented.
 - Added doctype to HTML files. CSS and JS inclusions have been ordered.
 - Fixed authors text.
 - Changes to pass linter.
 - Roles added as properties.
 - Client structure changed.
     - Moved bootstrap files to js folder.
     - Moved app.css to stylesheets folder.
 - Usage of forEach.
 - Usage of try-catch avoided.
 - Usage of constants.
 - Deleted proxy variable.
 - Other minor style fixes.
